### PR TITLE
Dev/systemd improvements

### PR DESCRIPTION
**Pull request by iliajie**
* Fix change bind page, including port redirects
 * Fix Webmin systemd service unit stability
 * Upgrade page needs more work

### CODE REVIEW

File: useradmin/md5-lib.pl [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 use strict;
 use warnings;
+no warnings 'redefine';
```
Comment (jcameron) [inline useradmin/md5-lib.pl:5]: This seems like it belongs in a different PR?

File: webmin/change_bind.cgi [hunk 2]
```diff
@@ -175,15 +175,19 @@ else { $url = $ENV{'SERVER_NAME'}; }
 if ($ENV{'HTTPS'} eq "ON") { $url = "https://$url"; }
 else { $url = "http://$url"; }
 
-if ($tconfig{'inframe'}) {
-	# Theme uses frames, so we need to redirect the whole frameset
+# Theme redirect if port changed
+if ($miniserv{'port'} ne $oldminiserv{'port'}) {
```
Comment (jcameron) [inline webmin/change_bind.cgi:179]: Should be != instead of ne

**Comment by iliajie** [comment]
Well, because (if I understand this correctly) it is simply impossible to have control over `SIGTERM` the old ways, as _miniserv_ no longer controlling the process, where `systemd` is in charge. As you call`/etc/webmin/stop` (even with `$SIG{'TERM'} = 'ignore';`) `miniserv` will just stop with `systemd` (as designed). Also, you cannot call `/etc/webmin/.stop-init` either, as it will mess with `systemd` perception of the process status.

I made it possible to have port change work smoothly. However, taking down miniserv with `stop` command won't just work in an expected way.

> Why do this in index.cgi ? It seems odd to force stop webmin here ..

If we do it the old way, the process will just die and no redirection will happen. 

I will keep looking more today in doing it in a better way.

**Comment by iliajie** [comment]
Alright, there is much better way of doing it - https://github.com/webmin/webmin/pull/1659/commits/f662edb8147911eb45f468ed8250c8daa097edd9.

Few more small fixes to this PR and done.

**Comment by iliajie** [comment]
Alright, unless you have questions, this can be merged. Works perfectly for me now on all distros!

File: init/updateboot.pl [hunk 3]
```diff
@@ -20,7 +20,14 @@
 			unlink("$p/$product.service");
 			unlink("$p/$product");
 			}
-		copy_source_dest("../webmin-systemd", "$systemd_root/$product.service");
+		my $temp = &transname();
+		&copy_source_dest("../webmin-systemd", "$temp");
```
Comment (jcameron) [inline init/updateboot.pl:24]: Use an absolute path rather than ..

File: web-lib-funcs.pl [hunk 4]
```diff
@@ -2401,6 +2401,13 @@ sub reload_miniserv
 {
 my ($ignore) = @_;
 return undef if (&is_readonly_mode());
+if (&has_command('systemctl')) {
+	my $restart_target =
```
Comment (jcameron) [inline web-lib-funcs.pl:2405]: Couldn't this just be `my $restart_target = &get_product_name();` ?

File: init/updateboot.pl (hunk 3 above)
Comment (jcameron) [reply init/updateboot.pl:24]: I still don't see why an absolute path like `$root_directory/webmin-systemd` can't be used here?

**Comment by iliajie** [comment]
> I still don't see why an absolute path like $root_directory/webmin-systemd can't be used here?

Yes, alright. Added it.

**Comment by iliajie** [comment]
> Couldn't this just be my $restart_target = &get_product_name(); ?

Yes, I could by I wanted to be super specific as this is going to system command.

File: init/updateboot.pl [hunk 5]
```diff
@@ -20,7 +20,14 @@
 			unlink("$p/$product.service");
 			unlink("$p/$product");
 			}
-		copy_source_dest("../webmin-systemd", "$systemd_root/$product.service");
+		my $temp = &transname();
+		&copy_source_dest("$root_directory/webmin-systemd", "$temp");
```
Comment (jcameron) [inline init/updateboot.pl:24]: $temp doesn't need to be in quotes here
